lib/ResourceSet.h

```diff
@@ -171,14 +191,14 @@ class ResourceSet
 
 // WARNING: comparison operators are used for "can afford" relation: a <= b means that foreach i a[i] <= b[i]
 // that doesn't work the other way: a > b doesn't mean that a cannot be afforded with b, it's still b can afford a
-// 		bool operator<(const ResourceSet &rhs)
-// 		{
-// 			for(int i = 0; i < size(); i++)
-// 				if(at(i) >= rhs[i])
-// 					return false;
-//
-// 			return true;
-// 		}
+ 	bool operator<(const ResourceSet &rhs)
+ 	{
+ 		for(int i = 0; i < size(); i++)
+			if (this->container.at(i) < rhs[i])
+ 				return true;
+
+ 		return false;
+ 	}
```

Comment:
Is this used for sorting? In that case it is not suitable as sorting function - a < b does not means that b > a.

If this is used for things like "can I afford this" then might be better to turn it into a method with clear name instead of operator overloading

Reply:
I checked, it's just for a check whether something can be afforded. For which a method "canAfford" already existed. Which I probably didn't know at the time. I'll just remove it and use the existing method instead.